cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -258,16 +258,16 @@ public synchronized void deleteDatasets(Id.Namespace namespaceId)
 
     if (checkProgramsRunning(namespaceId)) {
       throw new NamespaceCannotBeDeletedException(namespaceId,
-                                                  "Some programs are currently running in namespace " + namespaceId);
+                                                  String.format("Some programs are currently running in namespace " +
+                                                                  "'%s', please stop them before deleting datasets " +
```

Comment:
The API is to delete datasets -  the 'datasets' entity in CDAP. I think its fair here? Perhaps if we have something that is even more user facing than this (cli command/UI), we can make it data there?